ruma-client-api/src/r0/session/login.rs

```diff
@@ -53,11 +54,13 @@ ruma_api! {
         ///
         /// Will be the same as the corresponding parameter in the request, if one was
         /// specified.
-        pub device_id: Box<DeviceId>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub device_id: Option<Box<DeviceId>>,
```

Comment:
I don't think this is optional either, especially considering this line in the spec:

> If the client does not supply a device_id, the server must auto-generate one.

Reply:
See inhibit_login

Reply:
To me that's an indication that the `device_id` *can* be left out in the response if the corresponding request contained one.

Reply:
> See inhibit_login

I'm afraid you'll have to be a bit more verbose.

Reply:
> See inhibit_login

That's a parameter on `/register` which affects the response of that endpoint, but this PR is for `/login`.

Reply:
> To me that's an indication that the `device_id` _can_ be left out in the response if the corresponding request contained one.

We should ask for clarification, but this line makes me think that it's always required:

> ID of the logged-in device. Will be the same as the corresponding parameter in the request, if one was specified.